overlay/master.cpp

```diff
@@ -588,6 +602,11 @@ class Agent
         continue;
       }
 
+      // Skip if the overlay is disabled
+      if (!overlay->enabled) {
```

Comment:
Same comment as earlier. Let the system do its normal processing and depend on `flag` to take a call.